lib/migration/iiif.rb

```diff
@@ -33,16 +26,32 @@ def riiif
     def migrate_featured_images
       Spotlight::FeaturedImage.all.each do |image|
         update_iiif_url(image)
+        copy_exhibit_thumbnail_from_featured_image(image)
       end
     end
 
     def migrate_contact_avatars
       Spotlight::Contact.all.each do |contact|
         avatar = copy_contact_image_to_avatar(contact)
-        contact.update(avatar: avatar)
+        contact.update(avatar: avatar) if avatar
       end
     end
 
+    # Checks if the image was associated as a thumbnail.
+    # If so, this will update the STI type column of the FeaturedImage as well
+    # as copy the file over to the correct directory given the new class name
+    def copy_exhibit_thumbnail_from_featured_image(image)
+      return unless Spotlight::Exhibit.where(thumbnail_id: image.id).any?
+      filename = image.read_attribute_before_type_cast('image')
+      old_file = "public/#{image.image.store_dir}/#{filename}"
+      image.becomes!(Spotlight::ExhibitThumbnail)
+      image.save
+      # AR + STI seems to require that we re-query for this
+      # otherwise we get an association miss-match
+      reloaded_image = Spotlight::ExhibitThumbnail.find(image.id)
+      reloaded_image.image.store!(File.new(old_file))
+    end
+
     # Looks for a file at the old uploader location and copies it to a FeaturedImage
     def copy_contact_image_to_avatar(contact)
       filename = contact.read_attribute_before_type_cast('avatar')
```

Comment:
So, I believe the file path below is hard coded like this because we're trying to derive the CarrierWave storage directory from a class that does not exist anymore (an uploader was attached directly to the contact class, now that's handled through `FeaturedImage`).  Above, we can use the `store_dir` attribute safely because the classes are still present, but I'm not sure how we could derive the old classes w/o having some sort of hard coded path somewhere.

Totally up for suggestions on how to improve this though.